bundles/org.openhab.binding.ferroamp/src/main/java/org/openhab/binding/ferroamp/internal/FerroampHandler.java

```diff
@@ -0,0 +1,286 @@
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.State;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link FerroampHandler} is responsible for handling of the values sent to and from the binding.
+ *
+ * @author Örjan Backsell - Initial contribution
+ *
+ */
+@NonNullByDefault
+public class FerroampHandler extends BaseThingHandler implements MqttMessageSubscriber {
+    private final static Logger logger = LoggerFactory.getLogger(FerroampHandler.class);
+
+    private static @Nullable FerroampConfiguration ferroampConfig;
+    private @Nullable static MqttBrokerConnection ferroampConnection;
+    FerroampMqttCommunication ferroampMqttCommunication = new FerroampMqttCommunication(thing);
+
+    private List<FerroampChannelConfiguration> channelConfigEhub = new ArrayList<>();
+    private static List<FerroampChannelConfiguration> channelConfigSsoS0 = new ArrayList<>();
+    private static List<FerroampChannelConfiguration> channelConfigSsoS1 = new ArrayList<>();
+    private static List<FerroampChannelConfiguration> channelConfigSsoS2 = new ArrayList<>();
+    private static List<FerroampChannelConfiguration> channelConfigSsoS3 = new ArrayList<>();
+    private static List<FerroampChannelConfiguration> channelConfigEso = new ArrayList<>();
+    private static List<FerroampChannelConfiguration> channelConfigEsm = new ArrayList<>();
+
+    long refreshInterval = 30;
+    static boolean isEsoAvailable = false;
+    static boolean isEsmAvailable = false;
+
+    public FerroampHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        String transId = UUID.randomUUID().toString();
+        String valueConfiguration = command.toString();
+
+        if (FerroampBindingConstants.CHANNEL_REQUESTCHARGE.equals(channelUID.getId())) {
+            String requestCmdJsonCharge = "{\"" + "transId" + "\":\"" + transId
+                    + "\",\"cmd\":{\"name\":\"charge\",\"arg\":\"" + valueConfiguration + "\"}}";
+            sendMQTT(requestCmdJsonCharge);
+        }
+        if (FerroampBindingConstants.CHANNEL_REQUESTDISCHARGE.equals(channelUID.getId())) {
+            String requestCmdJsonDisCharge = "{\"" + "transId" + "\":\"" + transId
+                    + "\",\"cmd\":{\"name\":\"discharge\",\"arg\":\"" + valueConfiguration + "\"}}";
+            sendMQTT(requestCmdJsonDisCharge);
+        }
+        if (FerroampBindingConstants.CHANNEL_AUTO.equals(channelUID.getId())) {
+            String requestCmdJsonAuto = "{\"" + "transId" + "\":\"" + transId + "\",\"cmd\":{\"name\":\"auto\"}}";
+            sendMQTT(requestCmdJsonAuto);
+        }
+    }
+
+    @SuppressWarnings("null")
```

Comment:
20241107
Still ongoing.
I am not so familier with these things, so I am sorry, but I don't really understand your description

Reply:
This `@SuppressWarnings("null")` hides an actual null warning from the compiler. So the issue is still there, only the compiler does not complain about it anymore. This is wrong.
The issue should be fixed in the code instead of suppressing it.

When you start fixing these, you might encounter a situation that you cannot solve because the compiler or annotations are wrong, they make a fasle positive. In that case you can extra the line causing the issue to a very small private method that is annotated with this `@SuppressWarnings("null")`
Ny doing this, you make sure the suppression applies to only one line of code and not to the whole method.

Reply:
Thanks for the as always very good info. I will do my best to have it solved

Reply:
20241115
Tried to fix these things, Please see:
FerroampHandler.java
FerroampMqttCommunication.java